src/win/win_impl.rs

```diff
@@ -12,10 +20,10 @@ use std::mem::*;
 #[derive(Default)]
 pub struct Enigo;
 
-fn mouse_event(flags: u32, data: u32, dx: i32, dy: i32) {
-    let mut input = INPUT {
-        type_: INPUT_MOUSE,
-        u: unsafe {
+fn mouse_event(flags: MOUSE_EVENT_FLAGS, data: i32, dx: i32, dy: i32) {
+    let input = INPUT {
+        r#type: INPUT_MOUSE,
+        Anonymous: unsafe {
```

Comment:
Can we change this to:
```
let input = INPUT {
        r#type: INPUT_MOUSE,
        Anonymous: INPUT_0 {
            mi: MOUSEINPUT {
                dx,
                dy,
                mouseData: data,
                dwFlags: flags,
                time: 0,
                dwExtraInfo: 0,
            },
        },
    };
```
It compiles and the examples work. I am not familiar writing `unsafe` code though. Are there any reasons why `transmute` needs to be used here?

Reply:
Somehow I was unable to comment on the full `let input` statement